File: python/src/Structural/Flyweight/TeaFlavorFactory.py

#//TeaFlavorFactory.java - the Factory

from TeaFlavor import TeaFlavor

class TeaFlavorFactory :

	flavors = []
	teasMade = 0

	def getTeaFlavor(self,flavorToGet) :
			if self.teasMade > 0 :
				for flavor in self.flavors :
					if flavorToGet == flavor.getFlavor() :
						return flavor 
			retval = TeaFlavor(flavorToGet)
			self.flavors.append( retval )
			self.teasMade += 1
			return retval

	def getTotalTeaFlavorsMade(self) : 
		return self.teasMade

File: ruby/src/Structural/Flyweight/TeaFlavorFactory.rb

#//TeaFlavorFactory - the Factory

require_relative "TeaFlavor"

class TeaFlavorFactory

	def initialize()
		@flavors = []
		@teasMade = 0
	end

	def getTeaFlavor(flavorToGet)
		if @teasMade > 0
			for flavor in @flavors
				if flavorToGet == flavor.getFlavor()
					return flavor 
				end
			end
		end
		retval = TeaFlavor.new(flavorToGet)
		@flavors.push( retval )
		@teasMade += 1
		return retval
	end

	def getTotalTeaFlavorsMade()
 		return @teasMade
	end

end
